libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -5170,10 +5158,27 @@ void GCS_MAVLINK::convert_COMMAND_LONG_to_COMMAND_INT(const mavlink_command_long
     out.param2 = in.param2;
     out.param3 = in.param3;
     out.param4 = in.param4;
-    const bool stores_location = command_long_stores_location((MAV_CMD)in.command);
-    out.x = convert_COMMAND_LONG_loc_param(in.param5, stores_location);
-    out.y = convert_COMMAND_LONG_loc_param(in.param6, stores_location);
+
+    // we can't convert NaN/INF to an integer:
```

Comment:
We are supposed to deal with NaNs, it might be fine if the command is not using those params. We could convert from NaN which is no value for float Int32 max which is no value for int32 according to the spec.

Reply:
Oof.  Hmmm.  Will think about this more in the morning.

Problem is that if we just squash them to zero them we may end up triggering the sanitise-using-a-default-location code....

Reply:
The spec calls for INT32 max for invalid. (https://mavlink.io/en/messages/common.html#COMMAND_INT), so NaN should be converted to that. Then your new range check will bounce if it is a location.

Reply:
OK, squashed to uint32_t as you suggest - seems the best way forward here.

Note that this will be a behavioural change - previously we would have treated these as 0.